be/src/exec/pipeline/pipeline_builder.cpp

```diff
@@ -113,6 +120,8 @@ OpFactories PipelineBuilderContext::_maybe_interpolate_local_passthrough_exchang
         local_exchange = std::make_shared<AdaptivePassthroughExchanger>(mem_mgr, local_exchange_source.get());
     } else if (pass_through_type == LocalExchanger::PassThroughType::RANDOM) {
         local_exchange = std::make_shared<RandomPassthroughExchanger>(mem_mgr, local_exchange_source.get());
+    } else if (!state->is_writer_scale_closed() && pass_through_type == LocalExchanger::PassThroughType::SCALE) {
```

Comment:
`state->is_writer_scale_enabled() && pass_through_type == LocalExchanger::PassThroughType::SCALE` is more intuitive, I guess